install-dev/upgrade/php/ps_1770_update_charset.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+/**
+ * Update database to use utf8mb4 as a charset
+ */
+function ps_1770_update_charset() {
+    return Db::getInstance()->execute('ALTER DATABASE `' . _DB_NAME_ . '` CHARACTER SET = utf8mb4 COLLATE = utf8mb4_general_ci');
```

Comment:
What is the reason of doing this? It was already in 1.7.7.0.sql?

Reply:
Yes but it didn't work in the sql file because `_DB_NAME_` is not replaced automatically, nor can it be interpreted, so it tried to modify a database named `_DB_NAME_` which caused an error

Reply:
Oh I see, this is something I forget to add. I only do that for modules: https://github.com/PrestaShop/PrestaShop/pull/14894/files#diff-8f670e6ff5e6b010dcbb019a08eb69adR456

Should be update here: https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Install/Upgrade.php#L423

Reply:
mmm, nice! I'll add this modification and revert to a full SQL then 😉